### PR TITLE
Add support for compression

### LINKED ISSUE #5: Add Zlib support

### PR DESCRIPTION
**Pull request by Dreaded-Gnu**
Hello,

I found a possibility to easily integrate support for gzip and zlib compression by using `pako` as dependency.

Closes #5, #6

### CODE REVIEW

**Comment by eonarheim** [comment]
@Dreaded-Gnu I'll take a look today

File: src/index.ts [hunk 1]
```diff
@@ -178,15 +179,21 @@ export default class TiledResource extends Resource<ITiledMap> {
...
       for (var layer of data.layers) {
 
          if (typeof layer.data === "string") {
 
             if (layer.encoding === "base64") {
-               layer.data = decompressors.decompressBase64(<string>layer.data, layer.encoding);
+               layer.data = decompressors.decompressBase64(
+                  <string>layer.data,
+                  layer.encoding,
+                  typeof layer.compression !== "undefined"
```
Comment (kamranayub) [inline src/index.ts]: Any objections to using `layer.compression || ""` ?

**Comment by eonarheim** [comment]
Runs great locally

**Comment by kamranayub** [comment]
Yeah looked good to me! Thanks!

File: src/index.ts (hunk 1 above)
Comment (Dreaded-Gnu) [reply src/index.ts]: That's also a way, I'll change it.
I personally prefer the check against undefined.

**Comment by kamranayub** [comment]
👍

**Pull request merged**